runners/direct-java/build.gradle

```diff
@@ -40,9 +40,6 @@ applyJavaNature(
               include(project(path: it, configuration: "shadow"))
             }
           }
-          relocate "org.apache.beam.runners.core", getJavaRelocatedPath("runners.core")
```

Comment:
Would you mind explain a little bit this change? (what was the purposes of relocation here and why now it is no longer necessary, if possible)

Reply:
Yes it is a lot to put in a comment but let me know what you think should go in the code;

See the comment on line 22. It is complicated and actually wrong...

 -  Java-to-proto translations can only be registered once or they will cause a failure.
 -  The worry was: two runners could have registered different Java-to-proto translations and they would conflict, so the direct runner would relocate all of its registrations so the two runners could translate differently
    - it doesn't actually work this way (even with relocation you can't actually have two different translations really)
    - it would not have caused a conflict because the registrations in runners-core-construction would be the same AutoService class and not conflict
    - proto translation really should not exist differently per runner, but there are some hacks needed because of how we didn't put URNs in the core SDK, so runners have "translators" that are just there to give a URN to a non-serializable transform

Another annoying thing is that `org.apache.beam.runners.core` is execution-time utilities, but `org.apache.beam.runners.core.construction` was not. I think that partly this is accidentally catching it just because it has to be a package prefix. Honestly we could have just called it `construction` instead of `core.construction` and not had that problem.

Now the reason we _must_ get rid of it: after relocation, the registered `AvroCoderTranslatorRegistrar` is a subclass or `org.apache.beam.core.construction.CoderTranslatorRegistrar` but it is not a subclass of `org.apache.beam.runners.direct.relocated...org.apache.beam.core.construction.CoderTranslatorRegistrar`.

A possible alternative might have been to also include the avro extension and relocation into the direct runner shadowClosure but I don't want to add a dependency on avro.

Plus, this is temporary because after this PR it can all collapse into the core SDK and there is no more core-construction.

Reply:
Overall I really would love to not have any shadowClosure / relocation except for the vendored released artifacts. It builds faster and is way easier to debug and also intellij understands it better, etc.